src/shell.js

```diff
@@ -122,7 +122,7 @@ if (ENVIRONMENT_IS_NODE) {
 #endif // ENVIRONMENT_MAY_BE_NODE
 
 #if WASM_WORKERS
-var ENVIRONMENT_IS_WASM_WORKER = Module['$ww'];
+var ENVIRONMENT_IS_WASM_WORKER = !!Module['$ww'];
```

Comment:
This increases code size - is it necessary? Can we not fix the C method on the C side? (if not, perhaps rephrase the PR title/description)

Reply:
The C function is declared as returning `bool` which allows the compiler assume its going be be either a 1 or 0.  Anything else is undefined behaviour.

We could change the return type to it, but this fix seems better, since it also means that the `ENVIRONMENT_IS_WASM_WORKER` is only ever true or false which is is what its name suggests.  Storing the thread in this variable doesn't make much sense.

Reply:
Sorry, I'm not following. Perhaps my confusion is about `Module['$ww']`. I cannot find where in the code that is ever written to. Where is that done?

From the context of what you are saying, I am guessing that the output of `emscripten_current_thread_is_wasm_worker()` is written there. (Otherwise, how does this line connect to the topic of the PR?) Assuming my guess is right, then it should already be either 0 or 1, since it is a bool in C?

Reply:
I see now, thanks.